### PR TITLE
Add constraint

### PR DESCRIPTION
**Pull request by andrzejgorski**

### CODE REVIEW

File: whylog_vim/input_reader/__init__.py [hunk 1]
```diff
@@ -23,26 +21,32 @@ def parse_primary_key_groups(cls, content):
...
-                groups.append(
-                    (get_id_from_name(match.group(Groups.GROUP1)), int(match.group(Groups.GROUP2)))
-                )
+            if not are_params:
+                match = ConstraintInput.GROUP.match(line)
+                if match:
+                    groups.append(
+                        (
+                            get_id_from_name(match.group(Groups.GROUP1)),
+                            int(match.group(Groups.GROUP2))
```
Comment (ppolewicz) [inline whylog_vim/input_reader/__init__.py:44]: `match.group()` always returns an integer - cast is not necessary

File: whylog_vim/input_reader/__init__.py [hunk 2]
```diff
@@ -23,26 +21,32 @@ def parse_primary_key_groups(cls, content):
 
 
 class ConstraintReader(object):
-
-    CONSTRAINTS = {ConstraintType.IDENTICAL: IdenticalConstraint,}
-
     @classmethod
     def _parse_constraint(cls, lines):
         constr_type = ConstraintInput.TYPE.match(lines[0]).group(1)
```
Comment (ppolewicz) [inline whylog_vim/input_reader/__init__.py]: use `Groups.GROUP1` for consistency with the rest of the function

File: whylog_vim/input_reader/__init__.py [hunk 3]
```diff
@@ -23,26 +21,32 @@ def parse_primary_key_groups(cls, content):
 
 
 class ConstraintReader(object):
-
-    CONSTRAINTS = {ConstraintType.IDENTICAL: IdenticalConstraint,}
-
     @classmethod
     def _parse_constraint(cls, lines):
```
Comment (ppolewicz) [inline whylog_vim/input_reader/__init__.py:25]: logic in this function is difficult to comprehend without an example. Please add example input/output in docstring.

File: whylog_vim/output_formater/input_windows_messages.py [hunk 4]
```diff
@@ -58,17 +121,41 @@ def get_primary_key_message(cls, parser):
         output.add_commented(Messages.PRIMARY_KEY)
         cls._add_parser(output, parser)
         output.add_commented(Messages.ENDING)
+        output.add(', '.join([str(num) for num in parser.primary_key_groups]))
```
Comment (ppolewicz) [inline whylog_vim/output_formater/input_windows_messages.py]: allocating a list here is not necessary, so a generator expression is sufficient:

``` python
output.add(', '.join(str(num) for num in parser.primary_key_groups))
```

File: whylog_vim/output_formater/input_windows_messages.py [hunk 5]
```diff
@@ -58,17 +121,41 @@ def get_primary_key_message(cls, parser):
...
         output.add_commented(Messages.ENDING)
         return output
 
     @classmethod
+    def add_constraint(cls, parsers, constraint_name):
+        output = cls._get_constraint_message(parsers, WindowTypes.INPUT)
+        constraint = ConstraintRegistry.constraint_from_name(constraint_name)
+        output.add(ConstraintsOutputs.TYPE % constraint_name)
+        output.add(ConstraintsOutputs.GROUP % ('effect', '1'))
+        output.add(ConstraintsOutputs.GROUP % ('cause_1', '1'))
```
Comment (ppolewicz) [inline whylog_vim/output_formater/input_windows_messages.py]: is this a mock? if it is, please add an appropriate comment

File: whylog_vim/proxy/teacher/utils.py [hunk 6]
```diff
@@ -50,3 +53,40 @@ def set_parser_log_type(self, parser, log_type):
...
+    def write_constraint(self):
+        button_name = self.editor.get_button_name()
+        constraints = ConstraintRegistry.CONSTRAINTS.keys()
+        if button_name in constraints:
+            self.output = InputMessages.add_constraint(self.rule.parsers.values(), button_name)
+            self.main_proxy.create_input_window(self.output.get_content())
+            self.read_function = self.read_constraint
+        else:
+            six.print_('Wrong name')
+        return False
```
Comment (ppolewicz) [inline whylog_vim/proxy/teacher/utils.py]: why?

File: whylog_vim/proxy/teacher/utils.py [hunk 7]
```diff
@@ -50,3 +53,40 @@ def set_parser_log_type(self, parser, log_type):
...
+
+    def read_constraint(self):
+        constraint = ConstraintReader.create_constraint(self.editor.get_input_content())
+        self.teacher.register_constraint(self.get_next_constraints_id(), constraint)
+        return True
```
Comment (ppolewicz) [inline whylog_vim/proxy/teacher/utils.py]: why?

File: whylog_vim/input_reader/__init__.py (hunk 1 above)
Comment (andrzejgorski) [reply whylog_vim/input_reader/__init__.py:44]: For some reason which I don't understand it doesn't cast to integer.

Comment (ppolewicz) [reply whylog_vim/input_reader/__init__.py:44]: Yeah, I was wrong

**Pull request merged**